feat: populate new flow arrange-workspace-flow (#217)
griddynamics/rosetta: instructions/r3/core/workflows/arrange-workspace-flow-business-context.md

alwaysApply: false
tags: ["arrange", "workspace", "context", "phase"]
baseSchema: docs/schemas/phase.md
---

<arrange_workspace_business_context>

<description_and_purpose>
Interview user to gather non-technical and engineering behavior facts about the project not yet in `docs/CONTEXT.md`.
</description_and_purpose>

<phase_steps>
1. Interview user about what's missing in `docs/CONTEXT.md` — use `starter_topics`. Follow Questioning process in `hitl` SKILL.
2. Record confirmed answers to `docs/CONTEXT.md` per `context_contract`. Unconfirmed inferences → `docs/ASSUMPTIONS.md`.
3. Update `arrange-state.md`.
</phase_steps>

<starter_topics>
Starter topics (non-exhaustive — add project-specific questions as needed):
- Its overall goal.
- What it does in the client's wider ecosystem.
- The source and the target of the work.
- The issue tracker you use.
- How a story goes from ticket to implemented.
- Who the users and key stakeholders are.
- Core business rules and domain constraints.
- Any compliance or regulatory requirements.
- Accepted SDLC, DoD, and processes related to the project.
- References to documentation and ways to access it (example, acli or mcp for atlassian).
- etc.

> Example DoD: ACs fully met, all builds are without errors and warnings, changes covered with unit+integration+e2e tests at 85%+, database correct and migration present, backend starts and succeeds with expected responses for affected areas validated directly, frontend/mobile/app starts and affected areas fully functional validated directly, validation perform using manual QA by AI using fresh subagent.

Reviewer: ```suggestion

# Universal Starter Definition of Done

## Outcome

- [ ] Intent, ACs, and expected outcomes met
- [ ] User and business value proven end to end
- [ ] Scope complete with no known regressions

## Verification

- [ ] Build, lint, type, and static checks pass without errors or warnings
- [ ] Relevant unit + integration + E2E tests created and pass; changed code coverage ≥85%
- [ ] Affected components start and pass direct functional checks fully
- [ ] Fresh AI subagent completed direct manual QA and pass
- [ ] Security and privacy checks pass with no medium or higher findings
- [ ] Performance and reliability meet best practices
- [ ] UX and accessibility meet applicable standards
- [ ] Applicable laws, regulations, standards, policies, and licenses met
- [ ] Specific to project area, domain, technology best practices pass 

## Delivery

- [ ] Data and schema changes correct; migrations tested, safe, and reversible
- [ ] Dependencies, configuration, infrastructure, CI/CD, and deployment assets updated
- [ ] Observability, documentation, runbooks, and release notes updated
- [ ] Compatibility, deployment, rollback, and recovery verified

## Closure

- [ ] Known defects, risks, smells, and TODOs resolved or accepted and tracked
- [ ] Evidence linked and final acceptance recorded
```

Author: Thank you for the suggestion!
Done in [a232ed8](https://github.com/griddynamics/rosetta/pull/217/changes/a232ed8c3960213282da569bf9f5d6bd8857d39a) commit

---

> name: arrange-workspace-flow-business-context

Reviewer: Make AI to understand intent, not limit with findings, only, etc. 
Main goal: solicit information that is not known currently by AI (unknown unknowns) by asking different questions from user.
Provided list is a starter so that AI knows what we mean and where to dig.

Author: Done in the commit [099aa93](https://github.com/griddynamics/rosetta/pull/217/changes/099aa93ac96c9ac8a5f427c7be8e1e68a1403613)

description: "Phase 3 Business context of arrange-workspace-flow"
disable-model-invocation: true
user-invocable: false
alwaysApply: false
tags: ["arrangement", "workspace", "context", "phase"]
baseSchema: docs/schemas/phase.md
---

<arrange_workspace_business_context>

<description_and_purpose>
Close `CONTEXT.md` gaps, resolve contradictions, mine undocumented process — one continuous exploratory pass.
</description_and_purpose>

> <workflow_context>

Reviewer: Not needed => no subagents

Author: Done in the commit [7994b85](https://github.com/griddynamics/rosetta/pull/217/changes/7994b85faa11f6af3d40d9ccef42ab5fadf8dfa6)

2. Interview user on findings only
3. Update `CONTEXT.md`
</phase_steps>

<business_context_topics>
Read and record the non-technical and engineering behavior facts about the project in `docs/CONTEXT.md`:

- Its overall goal.
- What it does in the client's wider ecosystem.
- The source and the target of the work.
- The issue tracker you use.
- How a story goes from ticket to implemented.
- Who the users and key stakeholders are.
- Core business rules and domain constraints.
- Any compliance or regulatory requirements.
- Accepted SDLC, DoD, and processes related to the project.
- References to documentation and ways to access it (example, acli or mcp for atlassian).
- Example DoD: ACs fully met, all builds are without errors and warnings, changes covered with unit+integration+e2e tests at 85%+, database correct and migration present, backend starts and succeeds with expected responses for affected areas validated directly, frontend/mobile/app starts and affected areas fully functional validated directly, validation perform using manual QA by AI using fresh subagent.
</business_context_topics>

<find_findings step="3.2">

> 1. Read existing `docs/CONTEXT.md` if present

Reviewer: Workflow ran after init
What is the purpose of this phase? To put information that is NOT in the repo, make USER give that.

Author: Done as part of the commit [099aa93](https://github.com/griddynamics/rosetta/pull/217/changes/099aa93ac96c9ac8a5f427c7be8e1e68a1403613)

> 2. Compare existing `docs/CONTEXT.md` and repo evidence against `business_context_topics`.

Reviewer: Why?

Author: Removed in the commit [099aa93](https://github.com/griddynamics/rosetta/pull/217/changes/099aa93ac96c9ac8a5f427c7be8e1e68a1403613)

<interview_gaps step="3.3">

> 1. USE SKILL `hitl`.

Reviewer: Workflow already loaded, why do we need to load again?

Author: Fixed in the commit [6bc2949](https://github.com/griddynamics/rosetta/pull/217/changes/6bc2949ccabb051a52a25b6cc65c2ed6a90a993c)

<interview_findings step="3.3">

> 1. Interview only on findings (gaps, contradictions, tacit knowledge); skip fully covered topics.

Reviewer: ```suggestion
1. You can only skip fully covered topics.
```

Author: Thank you for pointing. I've applied it
